doc/man3/EVP_DigestInit.pod

```diff
@@ -182,17 +215,41 @@ EVP_get_digestbyobj()
 Returns an B<EVP_MD> structure when passed a digest name, a digest B<NID> or an
 B<ASN1_OBJECT> structure respectively.
 
+=item EVP_MD_CTX_pkey_ctx()
+
+Returns the B<EVP_PKEY_CTX> assigned to B<ctx>. The returned pointer should not
+be freed by the caller.
+
 =item EVP_MD_CTX_set_pkey_ctx()
 
 Assigns an B<EVP_PKEY_CTX> to B<EVP_MD_CTX>. This is usually used to provide
-a customzied B<EVP_PKEY_CTX> to L<EVP_DigestSignInit(3)> or
+a customized B<EVP_PKEY_CTX> to L<EVP_DigestSignInit(3)> or
 L<EVP_DigestVerifyInit(3)>. The B<pctx> passed to this function should be freed
 by the caller. A NULL B<pctx> pointer is also allowed to clear the B<EVP_PKEY_CTX>
 assigned to B<ctx>. In such case, freeing the cleared B<EVP_PKEY_CTX> or not
 depends on how the B<EVP_PKEY_CTX> is created.
 
 =back
 
+=head1 CONTROLS
+
+EVP_MD_CTX_ctrl() can be used to send the following standard controls:
+
+=over 4
+
+=item EVP_MD_CTRL_MICALG
+
+Sets the micalg string for S/MIME types.
+
+=item EVP_MD_CTRL_XOF_LEN
+
+This control sets the digest length for extendable output functions to B<p1>.
+Sending this control directly should not be necessary, the use of
+C<EVP_DigestFinalXOF()> is preferred.
+Currently used by SHAKE and KECCAK.
+
+=back
```

Comment:
Should we document `EVP_MD_CTRL_ALG_CTRL` here? I *think* the intent of this value is to be an offset for custom digest implementations to create their own non-standard controls.

Reply:
That could be right, but is it worth documenting? The only usage I could found was in the now removed [GOST implementation](https://github.com/openssl/openssl/commit/a4346646f12f83f8058abd9d013824fb90abfe73#diff-f2efce01b8619e51d81a951d0444af3aR160)

Reply:
It is still used by the external GOST engine:

https://github.com/gost-engine/engine/blob/6087586263eb1351a5fa370b233b8d747d0e1f59/gost_lcl.h#L246-L248

But I'm ok, if we don't want to document this.

Reply:
Leaving it out for now as it seems too niche.